rust/connlib/tunnel/src/device_channel/tun_windows.rs

```diff
@@ -73,6 +73,39 @@ impl Tun {
                 }
             };
 
+        let iface_idx = adapter.get_adapter_index()?;
+
+        // Remove any routes that were previously associated with us
+        // TODO: Pick a more elegant way to do this
+        Command::new("powershell")
+            .creation_flags(CREATE_NO_WINDOW)
+            .arg("-Command")
+            .arg(format!(
+                "Remove-NetRoute -InterfaceIndex {iface_idx} -Confirm:$false"
+            ))
+            .stdout(Stdio::null())
+            .status()?;
+
+        set_iface_config(adapter.get_luid(), DEFAULT_MTU)?;
+
+        let session = Arc::new(adapter.start_session(wintun::MAX_RING_CAPACITY)?);
+
+        let (packet_tx, packet_rx) = mpsc::channel(5);
+
+        let recv_thread = start_recv_thread(packet_tx, Arc::clone(&session))?;
+        let packet_rx = std::sync::Mutex::new(packet_rx);
+
+        Ok(Self {
+            adapter,
+            iface_idx,
+            _recv_thread: recv_thread,
+            packet_rx,
+            session: Arc::clone(&session),
+            routes: HashSet::new(),
+        })
+    }
+
+    pub fn set_config(&self, config: &InterfaceConfig, dns_config: &[IpAddr]) -> Result<()> {
```

Comment:
the only change in that file is separating `new` with `set_config` so it can be called twice

Reply:
and therefore removing the arguments from `new`